src/server/event.cpp

```diff
@@ -70,12 +72,26 @@ static void copy_recr(const recurrence_io_t &from, recurrence_pattern_t &to)
 
 bool event_t::check_attributes(string &error_message, const attribute_t &a, bool empty_only)
 {
-    static pcrecpp::RE known_keyword
-        = "APPLICATION|TITLE|TEST|COMMAND|USER|CROUP|"
-          "DBUS_SERVICE|DBUS_PATH|DBUS_INTERFACE|DBUS_METHOD|DBUS_SIGNAL|"
-          "PLUGIN|BACKUP";
-    static pcrecpp::RE upper_case = "[A-Z_]+";
-    static pcrecpp::RE app_name = "[A-Za-z_][A-Za-z_0-9]*";
+    static PCRE2_SPTR known_keyword = (PCRE2_SPTR) "APPLICATION|TITLE|TEST|COMMAND|USER|CROUP|"
+                                      "DBUS_SERVICE|DBUS_PATH|DBUS_INTERFACE|DBUS_METHOD|DBUS_SIGNAL|"
+                                                               "PLUGIN|BACKUP";
+    static PCRE2_SPTR upper_case = (PCRE2_SPTR) "[A-Z_]+";
+    static PCRE2_SPTR app_name = (PCRE2_SPTR) "[A-Za-z_][A-Za-z_0-9]*";
+
+    pcre2_code *known_keyword_re = pcre2_compile(known_keyword, PCRE2_ZERO_TERMINATED, 0, NULL, NULL, NULL);
```

Comment:
All these seem to be leaking now badly, there's nothing calling pcre2_code_free(). Besides that, earlier the instances were created only once while now each check creates their own ones.

Looks like pcre2 got missing this pcrecpp api that nicely handled these details earlier. Maybe some small c++ wrapper could be handling such locally?

OR maybe even rather take a step back: why is pcre used here in the first place? The 'known_keywords' just seems ensuring string is in a specific set. That doesn't need a regexp, QSet or std:: equivalent should be quite enough. The other ones could consider just helper methods or QRegularExpression. I don't see much of point on having any direct pcre/pcre2 dependency here especially if Qt is getting used.

Reply:
Sure I could convert all these to Qt gimme couple days (not because it will take long time, because i might not have time)